catalog/app/components/EChartsChart.tsx

```diff
@@ -0,0 +1,96 @@
+import * as echarts from 'echarts'
+import * as React from 'react'
+import * as R from 'ramda'
+import useResizeObserver from 'use-resize-observer'
+import * as M from '@material-ui/core'
+
+import { createBoundary } from 'utils/ErrorBoundary'
+import useMemoEq from 'utils/useMemoEq'
+
+export type DisposeHook = () => void
+export type InitHook = (chart: echarts.ECharts) => DisposeHook | void
+
+interface ChartInnerProps {
+  option: echarts.EChartsOption
+  resize?: boolean
+  theme?: Parameters<typeof echarts.init>[1]
+  initOptions?: Parameters<typeof echarts.init>[2]
+  onInit?: InitHook
+}
+
+function ChartInner({
+  option,
+  resize = true,
+  theme = undefined,
+  initOptions = undefined,
+  onInit = undefined,
+  ...props
+}: ChartInnerProps & M.BoxProps) {
+  const [error, setError] = React.useState<unknown>(null)
+  if (error) throw error
```

Comment:
> I think it will also produce another error: rendered less hooks than in the previous render

nope, it will just throw an error which will be propagated to a nearest error boundary